### ISSUE TITLE
3.0.0rc1: TypeHintWarning: Cannot resolve forward reference 'Inner'

### ISSUE THREAD

**Issue by mtelka** [agronholm/typeguard#303]
### Things to check first

- [X] I have searched the existing issues and didn't find my bug already reported there

- [X] I have checked that my bug is still present in the latest release

### Typeguard version

3.0.0rc1

### Python version

3.9.16

### What happened?

I ran tests for `3.0.0rc1`.  All tests passed, but the following warning was reported:
```
tests/test_instrumentation.py::test_inner_class_staticmethod[importhook]
tests/test_instrumentation.py::test_inner_class_classmethod[importhook]  
tests/test_instrumentation.py::test_inner_class_method[importhook]
tests/test_instrumentation.py::test_inner_class_classmethod[typechecked]
tests/test_instrumentation.py::test_inner_class_staticmethod[typechecked]
tests/test_instrumentation.py::test_inner_class_method[typechecked]      
  /usr/lib/python3.9/vendor-packages/typeguard/_checkers.py:625: TypeHintWarning: Cannot resolve forward reference 'Inner'
    warnings.warn(

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
```

### How can we reproduce the bug?

Running tests using tox should reproduce the problem.

**Comment by agronholm** [comment]
I'm aware of this. Is this a problem for you? I didn't consider it a showstopper issue.

**Comment by mtelka** [comment]
No, it is not a problem at all.  I just reported it in a hope it will get fixed in 3.0.0.  Thanks.

**Comment by agronholm** [comment]
I've documented the problem and added `xfail` tests to cover it.
